src/platforms/web/compiler/directives/model.js

```diff
@@ -26,19 +26,23 @@ export default function model (
     }
   }
   if (tag === 'select') {
-    genSelect(el, value)
+    genSelect(el, value, modifiers)
   } else if (tag === 'input' && type === 'checkbox') {
-    genCheckboxModel(el, value)
+    genCheckboxModel(el, value, modifiers)
   } else if (tag === 'input' && type === 'radio') {
-    genRadioModel(el, value)
+    genRadioModel(el, value, modifiers)
   } else {
     genDefaultModel(el, value, modifiers)
   }
   // ensure runtime directive metadata
   return true
 }
 
-function genCheckboxModel (el: ASTElement, value: string) {
+function genCheckboxModel (
+  el: ASTElement,
+  value: string,
+  modifiers: ?Object
```

Comment:
Is modifiers type `{[key: string]: true}` more specific?  Reference: https://github.com/vuejs/vue/blob/dev/flow/compiler.js#L46